cachecontrol/controller.py

```diff
@@ -58,8 +59,28 @@ def _urlnorm(cls, uri):
         return defrag_uri
 
     @classmethod
-    def cache_url(cls, uri):
-        return cls._urlnorm(uri)
+    def _authorization_digest(cls, authorization_header):
+        """Creates a hash digest of a request's authorization header.
+        """
+        if not authorization_header:
+            return ''
+        return hashlib.sha224(authorization_header.encode('utf-8')).hexdigest()
+
+    @classmethod
+    def cache_key(cls, request, response=None):
```

Comment:
I don't actually think we need the response (or even an `is_public` flag) anymore because we're using mget.

We can always just create both keys and return which ever one we find in the cache